mesonbuild/interpreter/type_checking.py

```diff
@@ -269,9 +269,9 @@ def _env_convertor(value: _FullEnvInitValueType) -> EnvironmentVariables:
 )
 
 # TODO: CustomTargetIndex should be supported here as well
-DEPENDS_KW: KwargInfo[T.List[T.Union[BuildTarget, CustomTarget]]] = KwargInfo(
+DEPENDS_KW: KwargInfo[T.List[T.Union[BuildTarget, CustomTarget, CustomTargetIndex]]] = KwargInfo(
     'depends',
-    ContainerTypeInfo(list, (BuildTarget, CustomTarget)),
+    ContainerTypeInfo(list, (BuildTarget, CustomTarget, CustomTargetIndex)),
     listify=True,
     default=[],
```

Comment:
```suggestion
    default=[],
    since_types={CustomTargetIndex: '1.5.0'},
```

Reply:
I guess, you meant `since_values`. This is added now.